src/main.h

```diff
@@ -1303,6 +1303,7 @@ class CBlockIndex
     // Gridcoin (17.6.2017 Brod) Add dpor related back-pointers
     CBlockIndex* ppCpid;  // previous block by same cpid
     CBlockIndex* ppSuper; // previous superblock
+    CSuperblock* pDesSuper; // deserialized superblock data

```

Comment:
That would be a brilliant solution. It would remove the need for memory optimizations for a very, very long time as the usage would be more or less constant. 55 megs for the chain instead of 280.